### DISCUSSION TITLE
Make deployments to trigger only if CI checks pass

### DISCUSSION THREAD

**Discussion by thunder-coding** [vercel/vercel#5776] [Ideas] [answered]
It would be great if deployments trigger only if CI tests pass.

Currently vercel deployments start as soon as a new commit is made in repository/PR.

For it may happen that the CI tests pass on PRs but fail on main branch. This will save a lot of husstles.

Nice to have: let user decide which checks are mandatory to pass for deployments to trigger.

**Comment by ImOnMars** [comment]
Agreed, I feel like this is a pretty essential feature

**Comment by leerob** [comment, marked as answer]
We have this on our roadmap!

**Reply by mejiasd3v** [reply]
hey @leerob, do you have any update on this?

**Reply by jonworek** [reply]
Any update @leerob? It's been a few years 😅

**Comment by dominikvrbic** [comment]
Are there any updates on this?

**Comment by humleflue** [comment]
Any updates on this?

**Comment by paulgould** [comment]
Bump!

**Comment by joshsny** [comment]
This would be great!
